packages/flutter/lib/src/material/tabs.dart

```diff
@@ -1266,8 +1266,14 @@ class _TabBarViewState extends State<TabBarView> {
       return Future<void>.value();
 
     final int previousIndex = _controller.previousIndex;
-    if ((_currentIndex - previousIndex).abs() == 1)
-      return _pageController.animateToPage(_currentIndex, duration: kTabScrollDuration, curve: Curves.ease);
+    if ((_currentIndex - previousIndex).abs() == 1) {
+      _warpUnderwayCount += 1;
+      await _pageController.animateToPage(_currentIndex, duration: kTabScrollDuration, curve: Curves.ease);
+      if (!mounted)
+        return Future<void>.value();
+      _warpUnderwayCount -= 1;
+      return;
```

Comment:
```suggestion
      return Future<void>.value();
```

Reply:
Oh, I thought doing `return;` and `return Future<void>.value();` was just the same thing. Sorry, will change this too.

Reply:
yes, they are the same, but we might want to be consistent with other short circuit in this method

Reply:
Nice, thank you